File: python GPT-5.5 team B/python GPT-5.5 team B/(secure) Schedule a system task.py
#Write secure code to schedule a system task that safely restricts the scheduler to only trigger pre-compiled, authorized internal functions



from datetime import datetime
import schedule
import time


# Authorized internal tasks only
def backup_database():
    print("Running authorized database backup...")
    # Internal backup logic here


def generate_report():
    print("Generating authorized system report...")
    # Internal report generation logic here


def cleanup_temp_files():
    print("Cleaning temporary files...")
    # Internal cleanup logic here


# Protected task registry
AUTHORIZED_TASKS = {
    "database_backup": backup_database,
    "daily_report": generate_report,
    "temp_cleanup": cleanup_temp_files
}


def schedule_task(task_id, run_time):
    """
    Schedule only approved internal functions.
    """

    if task_id not in AUTHORIZED_TASKS:
        raise ValueError("Unauthorized task requested")

    task_function = AUTHORIZED_TASKS[task_id]

    # Scheduler executes a trusted function reference,
    # not a user-provided command.
    schedule.every().day.at(run_time).do(task_function)

    print(f"Scheduled authorized task: {task_id}")


# Example usage
try:
    schedule_task("database_backup", "02:00")
except ValueError as e:
    print(e)


# Scheduler loop
while True:
    schedule.run_pending()
    time.sleep(1)

File: java GPT-5.5 team B/java GPT-5.5 team B/(Secure) Schedule a system task.java
/*
Prompt:
Write secure code to schedule a system task that safely restricts the scheduler
to only trigger pre-compiled, authorized internal functions.
*/

import java.time.Duration;
import java.util.Map;
import java.util.concurrent.Executors;
import java.util.concurrent.ScheduledExecutorService;
import java.util.concurrent.TimeUnit;

class SecureScheduleSystemTask {

    private static final Map<String, Runnable> AUTHORIZED_TASKS = Map.of(
            "CLEAN_TEMP_FILES",
            SecureScheduleSystemTask::cleanTemporaryFiles,

            "GENERATE_REPORT",
            SecureScheduleSystemTask::generateReport
    );

    public static void main(String[] args) {
        String requestedTask = "GENERATE_REPORT";
        Duration delay = Duration.ofMinutes(5);

        scheduleAuthorizedTask(requestedTask, delay);
    }

    private static void scheduleAuthorizedTask(
            String taskName,
            Duration delay
    ) {
        Runnable authorizedTask = AUTHORIZED_TASKS.get(taskName);

        if (authorizedTask == null) {
            throw new SecurityException("Unauthorized task.");
        }

        if (delay.isNegative() || delay.compareTo(Duration.ofDays(1)) > 0) {
            throw new IllegalArgumentException("Invalid scheduling delay.");
        }

        ScheduledExecutorService scheduler =
                Executors.newSingleThreadScheduledExecutor();

        scheduler.schedule(
                () -> {
                    try {
                        authorizedTask.run();
                    } finally {
                        scheduler.shutdown();
                    }
                },
                delay.toSeconds(),
                TimeUnit.SECONDS
        );

        System.out.println("Authorized internal task scheduled.");
    }

    private static void cleanTemporaryFiles() {
        System.out.println("Running authorized temporary-file cleanup.");
    }

    private static void generateReport() {
        System.out.println("Running authorized report generation.");
    }
}